### PR TITLE
Add minor changes

### PR DESCRIPTION
**Pull request by hitomi2500**
1) Added `user_early_init` function that, when overrided, allows user code execution when the SEGA logo is still on screen.
2) Added VDP1 HDTV flag, it allows switching VDP1 into HDTV (480p) modes
3) When setting cell format, `xxBMEN` flag in `CHCTLx` is reset now, allowing switching from bitmap to cell format.

### CODE REVIEW

File: libyaul/scu/bus/b/vdp/vdp1_env.c [hunk 1]
```diff
@@ -75,7 +75,7 @@ vdp1_env_set(const vdp1_env_t *env)
     _env_current_update(env);
 
     /* Always clear TVM and VBE bits */
-    _state_vdp1()->shadow_ioregs.tvmr = (env->rotation << 1) | env->bpp;
+    _state_vdp1()->shadow_ioregs.tvmr = (env->hdtv) ? 0x4 : (env->rotation << 1) | env->bpp;
```
Comment (ijacquez) [inline libyaul/scu/bus/b/vdp/vdp1_env.c:78]: Can you add parentheses around the ternary expression to be clearer?

File: libyaul/kernel/sys/init.c [hunk 2]
```diff
@@ -35,6 +37,8 @@ __sys_init(void)
 
     _bss_clear();
 
+    _early_access();
```
Comment (ijacquez) [inline libyaul/kernel/sys/init.c:40]: Just for clarity, could you rename this function to `user_early_init`?

File: libyaul/kernel/sys/init.c [hunk 3]
```diff
@@ -25,6 +25,8 @@ void user_init(void) __weak;
 
 static void _bss_clear(void);
 
+void __weak _early_access() {;}
```
Comment (ijacquez) [inline libyaul/kernel/sys/init.c:28]: You can define it how `user_init` is defined. Also, the prototype is missing in `user.h` file.

Comment (hitomi2500) [reply libyaul/kernel/sys/init.c:28]: Regarding `user.h`, should i use the same comment message as one used for `user_init`? Or should i add some extra text that explains the `user_early_init` purpose?

Comment (ijacquez) [reply libyaul/kernel/sys/init.c]: As is tradition with Yaul, you can copy the same comment :)

Comment (ijacquez) [reply libyaul/kernel/sys/init.c]: I forgot to mention that checking for `NULL` is also necessary.

File: libyaul/kernel/sys/init.c [hunk 4]
```diff
@@ -37,7 +37,9 @@ __sys_init(void)
 
     _bss_clear();
 
-    _early_access();
+    if (user_early_init != NULL) {
+        user_early_init();
```
Comment (ijacquez) [inline libyaul/kernel/sys/init.c:41]: I've attached a test that shows that this does indeed work. See [`dbgio-font.zip`](https://github.com/user-attachments/files/18055023/dbgio-font.zip).

This isn't required for this PR, but I move `user_early_init` way down past `__vdp_init`. But this required some changes on my end.

The motivation behind moving it further down is that the way it's implemented in this PR means that C++ would be broken. Specifically, if static objects exist and are referenced in `user_early_init`, their constructors wouldn't be called until much later. Static initialization is called before `main` or in our case, `user_init`.

File: libyaul/scu/bus/b/vdp/vdp1_env.c (hunk 1 above)
Comment (ijacquez) [reply libyaul/scu/bus/b/vdp/vdp1_env.c]: The only concern here is that if bit 2 is set, then other two bits should be cleared. I'm addressing this after this gets merged in.

**Comment by ijacquez** [comment]
Merged. Thank you for the fixes.

File: libyaul/kernel/sys/init.c (hunk 4 above)
Comment (hitomi2500) [reply libyaul/kernel/sys/init.c:41]: The whole point of early user init is to kick in before `__vdp_init`, and the fact that constructors were not called yet should be taken into consideration while implementing it. Frankly, it sounds dirty, and i'm sorry for this pollution. But moving every constructor before `__vdp_init` looks like a lot of work, and i'm not sure it's worth it. Maybe creating a dedicated example with corresponding comments is a reasonable compromise?

**Pull request merged**